src/main/io/gps.c

```diff
@@ -1560,6 +2078,7 @@ static union {
     ubxNavSvinfo_t svinfo;
     ubxNavSat_t sat;
     ubxCfgGnss_t gnss;
+    ubxMonVer_t ver;
     ubxAck_t ack;
     uint8_t bytes[UBLOX_PAYLOAD_SIZE];
```

Comment:
`uint8_t bytes[];`

Reply:
If I try that, I get a warning `2076:13: error: flexible array member in union`

Reply:
Well, sometimes this part of C standard is pretty annoying.
Keep is as it is now or use `uint8_t bytes[0]` 

Possible alternatives:

- uint8_t bytes[0] - this is good choice, assuming that BF does not use `gcc -pedantic`. 

- `#define UBLOX_PAYLOAD_SIZE MAX(sizeof(ubxNavPosllh_t), MAX(sizeof(ubxNavStatus_t), .... )))`
but that is ugly and long. boost-pp may be used to simplify it a bit (#define MAX_N variadic macro).

- define _buffer as `union { union ubxMessages_u m; uint8_t bytes[sizeof(union ubxMessages_u] } _buffer;`. It will affect all code using _buffer (`_buffer` -> `_buffer.m`), with little gain

Reply:
It seems like very little gain as `UBLOX_PAYLOAD_SIZE` (392) is comparable to but larger than `ubxMonVer_t` (340) - or am I missing something, @ledvinap?

Reply:
The idea was to take max of all members somehow. Assuming that SAT message is largest one is pretty fragile (For example, we can significantly reduce number of processed sats).

With  `uint8_t bytes[0]` we can `#define UBLOX_PAYLOAD_SIZE sizeof(_buffer)` and just avoid using `sizeof(_buffer.bytes)` in the code

Reply:
The Sat message is not necessarily the largest, certainly on the early stages of the connection where there are only four or less sats in view.

Please let me know *exactly* what code changes are needed, or perhaps leave it for followup PR?

Reply:
@ctzsnooze: Allocated size (32 sat fragments) is important, not actual message size.

Don't worry about it now, it can be easily improved later